hack/aad.sh

```diff
@@ -18,55 +25,75 @@ app-create)
     AZURE_AAD_CLIENT_SECRET=$(uuidgen)
     AZURE_AAD_CLIENT_ID=$(az ad app create \
         --display-name "$2" \
-        --homepage "https://$2/" \
-        --identifier-uris "https://$2/" \
+        --homepage "$3" \
+        --identifier-uris "$3" \
         --key-type password \
         --password "$AZURE_AAD_CLIENT_SECRET" \
         --query appId \
-        --reply-urls "https://$2/oauth2callback/Azure%20AD" \
+        --reply-urls "$3" \
         --required-resource-accesses @- <<'EOF' | tr -d '"'
 [
     {
-        "resourceAppId": "00000002-0000-0000-c000-000000000000",
-        "resourceAccess": [
-            {
-                "id": "311a71cc-e848-46a1-bdf8-97ff7156d8e6",
-                "type": "Scope"
-            }
-        ]
+      "resourceAppId": "00000003-0000-0000-c000-000000000000",
+      "resourceAccess": [
+        {
+          "id": "7ab1d382-f21e-4acd-a863-ba3e13f7da61",
+          "type": "Role"
+        },
+        {
+          "id": "5f8c59db-677d-491f-a6b8-5f174b11ec1d",
+          "type": "Scope"
+        },
+        {
+          "id": "5b567255-7703-4780-807c-7be8301ae99b",
+          "type": "Role"
+        },
+        {
+          "id": "37f7f235-527c-4136-accd-4a02d197296e",
+          "type": "Scope"
+        }
+      ]
+    },
+    {
+      "resourceAppId": "00000002-0000-0000-c000-000000000000",
+      "resourceAccess": [
+        {
+          "id": "311a71cc-e848-46a1-bdf8-97ff7156d8e6",
+          "type": "Scope"
+        }
+      ]
     }
 ]
 EOF
 )
 
-    az ad sp create --id $AZURE_AAD_CLIENT_ID >/dev/null
-
-    success=
-    for ((i=0; i<12; i++)); do
-        if az role assignment create -g $3 --assignee $AZURE_AAD_CLIENT_ID --role contributor &>/dev/null; then
-            success=true
-            break
-        fi
-        sleep 5
-    done
-    if [[ -z "$success" ]]; then
-        echo 'error: failed to assign contributor role to SP' >&2
-        exit 1
-    fi
-
     cat <<EOF
+AZURE_AAD_APP_NAME=$2
```

Comment:
suggest that here you just output AZURE_AAD_CLIENT_ID to stdout, and if you want to output an end user hint, do that to stderr